airflow/utils/log/colored_log.py

```diff
@@ -89,6 +89,9 @@ def _color_record_traceback(self, record: LogRecord) -> LogRecord:
         return record
 
     def format(self, record: LogRecord) -> str:
-        record = self._color_record_args(record)
-        record = self._color_record_traceback(record)
-        return super().format(record)
+        try:
+            record = self._color_record_args(record)
+            record = self._color_record_traceback(record)
+            return super().format(record)
+        except ValueError:  # I/O operation on closed file
+            return Formatter().format(record)
```

Comment:
Should we import the `Formatter` here? So it will only import if really required.